### PR TITLE
Mark NeighborParticles as a 'selfTest'

### PR DESCRIPTION
**Pull request by atmyers**
The proposed changes:
- [ ] fix a bug or incorrect behavior in AMReX
- [ ] add new capabilities to AMReX
- [ ] changes answers in the test suite to more than roundoff level
- [ ] are likely to significantly affect the results of downstream AMReX users
- [ ] include documentation in the code and/or rst files, if appropriate

### CODE REVIEW

**Review by jmsexton03 — APPROVED** [review]

File: Tools/RegressionTesting/AMReX-cuda-tests.ini [hunk 1]
```diff
@@ -359,6 +359,7 @@ useMPI = 1
 numprocs = 1
 useOMP = 0
 compileTest = 0
+selfTest = 1
 doComparison = 0
 diffDir = neighbor_particles.0
```
Comment (WeiqunZhang) [inline Tools/RegressionTesting/AMReX-cuda-tests.ini]: Do we still need `diffDir`?

Comment (atmyers) [inline Tools/RegressionTesting/AMReX-cuda-tests.ini]:
```suggestion
```

File: Tools/RegressionTesting/AMReX-hip-tests.ini [hunk 2]
```diff
@@ -357,6 +357,7 @@ useMPI = 0
 numprocs = 1
 useOMP = 0
 compileTest = 0
+selfTest = 1
 doComparison = 0
 diffDir = neighbor_particles.0
```
Comment (atmyers) [inline Tools/RegressionTesting/AMReX-hip-tests.ini]:
```suggestion
```

File: Tools/RegressionTesting/AMReX-cuda-tests.ini (hunk 1 above)
Comment (atmyers) [reply Tools/RegressionTesting/AMReX-cuda-tests.ini]: I think it will be ignored, but no we don't need it any more.

Comment (atmyers) [reply Tools/RegressionTesting/AMReX-cuda-tests.ini]: I think it will be ignored, but no we don't need it any more.

**Pull request merged**